src/javascript/app_2/Stores/Modules/Trading/trade-store.js

```diff
@@ -334,13 +334,20 @@ export default class TradeStore extends BaseStore {
             this.currency = obj_new_values.currency;
         }
 
+        let has_only_forward_starting_contracts;
+
+        if (/symbol/.test(Object.keys(obj_new_values))) {
+            await Symbol.onChangeSymbolAsync(obj_new_values.symbol);
+            has_only_forward_starting_contracts =
+                ContractType.getContractCategories().has_only_forward_starting_contracts;
+        }
+        this.root_store.ui.setHasOnlyForwardingContracts(has_only_forward_starting_contracts);
```

Comment:
LGTM, think we can add a `// TODO: remove this once future contracts are implemented`?  I'm a bit concerned that this couples the trade-store with the ui-store but I can't think of a better way 😄

Reply:
yes I'll add a comment